File: api/Models/Position.cs
namespace FMCK.Trainer.Api.Models;

using System.Text.Json.Serialization;

public class Position
{
    [JsonPropertyName("id")]
    public Guid Id { get; set; }
    
    [JsonPropertyName("name")]
    public string Name { get; set; } = string.Empty;

    [JsonPropertyName("coordinates")]
    public Coordinates Coordinates { get; set; } = new();

    [JsonPropertyName("address")]
    public string Address { get; set; } = string.Empty;
}

public class Coordinates
{
    [JsonPropertyName("lat")]
    public double Lat { get; set; }

    [JsonPropertyName("lon")]
    public double Lon { get; set; }
}

File: app/src/types/position.ts
export type Coordinates = {
	lat: number
	lon: number
}

export type Sweref99Coordinates = {
	northing: number
	easting: number
}

export type Position = {
	id: string
	name: string
	coordinates: Coordinates
	sweref99Coordinates: Sweref99Coordinates
	address: string
}
